### PR TITLE
allow to download email addresses

### PR DESCRIPTION
**Pull request by RamRamez**
## Summary by CodeRabbit

- **New Features**
	- Introduced the ability to export project-related email addresses to a CSV file.
	- Added a new action for exporting emails in the AdminJS interface, accessible to authorized users.
	- Created a new endpoint for downloading files directly from the admin panel.

- **Enhancements**
	- Expanded the role-based permissions system to include email export capabilities for specific user roles.

### CODE REVIEW

File: src/server/adminJs/tabs/projectsTab.ts [hunk 1]
```diff
@@ -371,6 +373,46 @@ export const updateStatusOfProjects = async (
...
+
+export const exportEmails = async (
+  context: AdminJsContextInterface,
+  request: AdminJsRequestInterface,
+) => {
+  const { records, currentAdmin } = context;
+  const projectIds = request?.query?.recordIds
+    ?.split(',')
+    ?.map(strId => Number(strId)) as number[];
+  const projects = await Project.createQueryBuilder('project')
```
Comment (ae2079) [inline src/server/adminJs/tabs/projectsTab.ts:392]: Maybe using Project.query has better performance

File: src/server/adminJs/tabs/projectsTab.ts [hunk 2]
```diff
@@ -371,6 +373,46 @@ export const updateStatusOfProjects = async (
...
+    .where('project.id IN (:...ids)', { ids: projectIds })
+    .getMany();
+  const data = projects.map(p => ({
+    title: p.title,
+    email: p.adminUser.email,
+  }));
+  const csv = convertToCSV(data);
+  const filePath = path.join(__dirname, 'exports', 'emails.csv');
+  fs.mkdirSync(path.dirname(filePath), { recursive: true });
+  fs.writeFileSync(filePath, csv);
```
Comment (ae2079) [inline src/server/adminJs/tabs/projectsTab.ts:405]: Do you consider the scenario that the file exists from before?
(Does it append the data or over write them?)

**Review by ae2079 — COMMENTED** [review]
Thanks @RamRamez 
I'm just adding some suggestions

File: src/server/adminJs/tabs/projectsTab.ts (hunk 2 above)
Comment (RamRamez) [reply src/server/adminJs/tabs/projectsTab.ts:405]: Yes, it will overwrite

File: src/server/adminJs/tabs/projectsTab.ts (hunk 1 above)
Comment (RamRamez) [reply src/server/adminJs/tabs/projectsTab.ts:392]: It's not a cronjob and it will be triggered only by comms team

**Review by ae2079 — APPROVED** [review]
Thanks @RamRamez 
LGTM

**Pull request merged**